### PR TITLE
Fix invalid handling of certain hash schemes in dovecot method

### LINKED ISSUE #7362: 'dovecot' method broken in password plugin for some hash schemes
When using the 'dovecot' method in the password plugin, it complains that the encryption function was not found, when using the argon2i (and likely certain other) hash schemes.

The problem lies in the preg_match comparison, which simply checks for false, which is an incorrect usage, according to the php documentation, as it can return a value that evaluates to false, even if a match was found.

A viable and likely more efficient solution is a call to mb_strpos, with a comparison to 0, since the method should be at the start of the string.

`if(!(mb_strpos($crypted,'{' . $method . '}') === 0)) {
                return false;
            }`

Pull request with this fix will be in shortly.

### PR DESCRIPTION
**Pull request by dodexahedron**
This resolves #7362 

The comparison to false is invalid, in the original function, because preg_match can return a false-equivalent value when a match was, in fact, made.

### CODE REVIEW

File: plugins/password/password.php [hunk 1]
```diff
@@ -712,6 +712,8 @@ static function hash_password($password, $method = '', $prefixed = true)
                 $method = 'CRAM-MD5';
             }
 
+            $method = strtoupper($method);
```
Comment (dodexahedron) [inline plugins/password/password.php:715]: This is not directly related to the bug, but it does allow this configuration directive to be case-insensitive, which seems desirable.

**Comment by alecpl** [comment]
It does not make sense to me. "preg_match() returns 1 if the pattern matches given subject, 0 if it does not, or FALSE if an error occurred."

What is your real case that didn't work? What's in `$method` and what's in `$crypted`?

**Comment by dodexahedron** [comment]
As described in the issue, argon2i fails reliably, for any input text.
It also fails if it is provided in all caps as ARGON2I, in the configuration.

**Comment by alecpl** [comment]
I don't see how is that possible and I don't have dovecot installed. So, it would help if I know a sample `$crypted` value that causes issues.

What PHP version?

**Comment by dodexahedron** [comment]
PHP 7.4
Honestly not sure what frame of mind I was in, last night...
In any case, I've extracted that code out of the method and am testing it in isolation to see what happened and why.
The behavior did not make sense to me, either, but the strpos change fixed it, last night.
I'm also closing the pull request until I've figured it out with more certainty. I'll leave the issue open.
